org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationMatching.cs

```diff
@@ -0,0 +1,718 @@
+                    person = personService.Get(matchingIds.First());
+                    // New single match found, increment the counter!
+                    idSet++;
+                }
+
+                Debug.WriteLine("Matching People Found: " + matchingIds.ToString());
+                WriteToLog(string.Format("    After SQL: {0}", watch.ElapsedMilliseconds));
+
+                // Atempt to match 1:1 based on email history making sure we exclude emails with multiple matches in the person table
+                if ( person == null && !hasMultiEmail )
+                {
+                    string email = contact.properties.email.ToLower();
+                    var matches = personSearchKeyService.Queryable().Where(k => k.SearchTypeValueId == 3497 && k.SearchValue == email).Select(k => k.PersonAlias.PersonId).Distinct().ToList();
+
+                    if ( matches != null )
+                    {
+                        Debug.WriteLine("PersonSearchKeyService match count: " + matches.Count());
+                        if ( matches.Count() == 1 )
+                        {
+                            // If 1:1 Email match and Hubspot has no other info, make it a match
+                            if ( String.IsNullOrEmpty(contact.properties.firstname) && String.IsNullOrEmpty(contact.properties.lastname) )
+                            {
+                                person = personService.Get(matches.First());
+                                // New single match found, increment the counter!
+                                idSet++;
+                            }
+                        }
+                    }
+
+                }
+                WriteToLog(string.Format("    After Email History: {0}", watch.ElapsedMilliseconds));
+
+
+                // Try to mark people that are potential matches, only people who can at least match email or phone and one other thing, only once
+                bool inBucket = false;
+                if (person == null && contact.properties.has_potential_rock_match != "True")
+                {
+                    // Matches phone number and one other piece of info
+                    if (!String.IsNullOrEmpty(contact.properties.phone))
+                    {
+                        var phone_matches = personService.Queryable().Where(p => p.PhoneNumbers.Select(pn => pn.Number).Contains(contact.properties.phone)).ToList();
+                        if (phone_matches.Count() > 0)
+                        {
+                            phone_matches = phone_matches.Where(p => CustomEquals(p.FirstName, contact.properties.firstname) || CustomEquals(p.NickName, contact.properties.firstname) || CustomEquals(p.Email, contact.properties.email) || CustomEquals(p.LastName, contact.properties.lastname)).ToList();
+                            for (var j = 0; j < phone_matches.Count(); j++)
+                            {
+                                //Save this information in the excel sheet....
```

Comment:
Informational comments like these should have a space after the slashes. Can you update this and similar comments?